examples/dataflow-data-generator/data-generator-pipeline/data_generator/ParquetUtil.py

```diff
@@ -25,30 +25,56 @@ def get_pyarrow_translated_schema(string_schema):
     :param string_schema:
     :return: pyarrow schema
     """
-    type_conversions = {
-        'STRING': pa.string(),
-        'NUMERIC': pa.int64(),
-        'BYTE': None,
-        'INTEGER': pa.int64(),
-        'FLOAT': pa.float64(),
-        'NUMERIC': pa.int64(),
-        'BOOLEAN': pa.bool_(),
-        'TIMESTAMP': pa.timestamp('us'),
-        'DATE': pa.date32(),
-        'TIME': pa.time64('us'),
-        'DATETIME': pa.timestamp('us'),
-        'GEOGRAPHY': None,
-        'RECORD': None
-    }
+    def _bq_to_pa_type(field):
+        """
+        A function to convert BigQuery types to pyarrow types.
+        :param field (bigquery.schema.SchemaField)
+        :return: pa.DataType
+        """
+        type_conversions = {
+            'STRING': pa.string(),
+            'NUMERIC': pa.int64(),
+            'BYTE': None,
+            'INTEGER': pa.int64(),
+            'FLOAT': pa.float64(),
+            'NUMERIC': pa.int64(),
+            'BOOLEAN': pa.bool_(),
+            'TIMESTAMP': pa.timestamp('us'),
+            'DATE': pa.date32(),
+            'TIME': pa.time64('us'),
+            'DATETIME': pa.timestamp('us'),
+            'GEOGRAPHY': None,
+            'RECORD': None
+        }
+
+        if field['mode'] == 'REPEATED':
+            if field['type'] == 'RECORD':
+                nested_fields = field['fields']
+                # Recursively call to convert the next nested layer.
+                return pa.list_(pa.struct(
+                    [(fld['name'], _bq_to_pa_type(fld))
+                        for fld in nested_fields]))
+            else:
+                return pa.list_(
+                    _bq_to_pa_type(type_conversions.get(field['type'])))
+        elif field['type'] == 'RECORD':
+            nested_fields = field['fields']
+            # Recursively call to convert the next nested layer.
+            return pa.struct(
+                [(fld['name'], _bq_to_pa_type(fld))
+                    for fld in nested_fields])
+        else:
+            return type_conversions.get(field.get('type'))
```

Comment:
Just curious why you used the get() method on the dict here rather than dict[key] method (type_conversions[field['type']]) since the dict[key] method is used elsewhere

Reply:
Good catch. Using `dict.get` returns `None` if the key doesn't exist while `dict[key]` will throw a `KeyError`. If it gets here and the type is not present in `type_conversions I should raise an exception.